dev/run-python-flavor-tests.sh

```diff
@@ -1,23 +1,15 @@
 #!/usr/bin/env bash
 set -x
-# Set err=1 if any commands exit with non-zero status as described in
-# https://stackoverflow.com/a/42219754
-err=0
-trap 'err=1' ERR
-export MLFLOW_HOME=$(pwd)
 
-# Run ML framework tests in their own Python processes to avoid OOM issues due to per-framework
-# overhead
-pytest tests/azureml
-pytest tests/utils/test_model_utils.py
+export MLFLOW_HOME=$(pwd)
 
 # TODO: Run tests for h2o, shap, and paddle in the cross-version-tests workflow
-pytest tests/h2o
-pytest tests/shap
-pytest tests/paddle
-
-pytest tests/tracking/fluent/test_fluent_autolog.py
-pytest tests/autologging
-pytest tests/test_mlflow_lazily_imports_ml_packages.py --lazy-import
-
-test $err = 0
+pytest \
+  tests/azureml \
+  tests/utils/test_model_utils.py \
+  tests/h2o \
+  tests/shap \
+  tests/paddle \
+  tests/tracking/fluent/test_fluent_autolog.py \
+  tests/autologging \
+  tests/test_mlflow_lazily_imports_ml_packages.py
```

Comment:
This change allows us to quickly find tests that failed.

Reply:
Let's say one of the tests in `tests/azureml` failed.

Before: we need to scroll back to the output log of `pytest tests/azureml`.
After: The output log looks like below:

```
...
============================= slowest 10 durations =============================
85.80s call     tests/h2o/test_h2o_model_export.py::test_pyfunc_serve_and_score
52.98s call     tests/shap/test_log.py::test_pyfunc_serve_and_score
45.46s call     tests/paddle/test_paddle_model_export.py::test_pyfunc_serve_and_score
12.01s call     tests/autologging/test_autologging_behaviors_integration.py::test_autolog_respects_silent_mode
11.58s call     tests/shap/test_log.py::test_log_model_with_extra_pip_requirements
10.74s call     tests/shap/test_log.py::test_sklearn_log_explainer
10.27s call     tests/shap/test_shap.py::test_log_explanation_with_classifier
9.02s setup    tests/shap/test_shap.py::test_log_explanation_with_classifier
8.40s call     tests/h2o/test_h2o_model_export.py::test_signature_and_examples_are_saved_correctly
8.09s call     tests/paddle/test_paddle_autolog.py::test_autolog_logs_expected_data
=========================== short test summary info ============================
FAILED tests/azureml/test_xxx.py::test_yyy
========== 1 failed, 1056 passed, 6189 warnings in 554.31s (0:09:14) ===========
```

Reply:
Good improvement!